### PR TITLE
Fix setup on MAC OSX

### PR DESCRIPTION
**Pull request by vsmalladi**
Fixes issues #74 for MAC OSX

### CODE REVIEW

**Comment by dhimmel** [comment]
@vsmalladi it would probably be best to just have a single command that works on both linux and macOS. Did you test those commands on Linux? If not, I can try them out.

**Comment by vsmalladi** [comment]
@dhimmel I haven't yet. If you can verify they work on Linux, then I can update the file to only have one set of commands.

**Comment by agitter** [comment]
I tested this in Linux and the Git Bash client for Windows.  In both cases `sed -i '' "s/greenelab/$OWNER/g" README.md` gave me the error `sed: can't read s/greenelab/agitter/g: No such file or directory`.  The command ` sed --in-place "s/greenelab/$OWNER/g" README.md` worked.

**Comment by dhimmel** [comment]
It looks like it [will be hard](https://unix.stackexchange.com/questions/13711/differences-between-sed-on-mac-osx-and-other-standard-sed) to find cross-platform compatible `sed` commands. Let me look into possible python one liners

File: SETUP.md [hunk 1]
```diff
@@ -137,6 +147,16 @@ sed --in-place "s/manubot-rootstock/$REPO/g" README.md
 git rm content/02.delete-me.md
 ```
 
+For setup on MAC OSX using BSD sed use the following commands.
+
+```sh
+sed -i '' "s/greenelab/$OWNER/g" README.md
```
Comment (dhimmel) [inline SETUP.md]: What if we abandoned `--inplace`/`-1` and just ran a subseqent `mv` command?

File: SETUP.md [hunk 2]
```diff
@@ -104,6 +104,16 @@ TRAVIS_ENCRYPT_ID=`grep \
 sed --in-place "s/f2f00aaf6402/$TRAVIS_ENCRYPT_ID/g" deploy.sh
 ```
 
+For setup on MAC OSX using BSD sed use the following commands.
```
Comment (dhimmel) [inline SETUP.md]: We could try a python command like:

```sh
python -c "import re; print(re.search('encrypted_([a-zA-Z0-9]+)_key', open('README.md').read())).group(0)"
```

Not tested BTW.

File: SETUP.md (hunk 1 above)
Comment (vsmalladi) [reply SETUP.md]: We could have the first sed command output to tmp file and the send one take tmp and output into README.md, then follow with rm for tmp file.

Comment (dhimmel) [reply SETUP.md]: If you use `mv -f`, you can skip the `rm` command.

File: SETUP.md (hunk 2 above)
Comment (vsmalladi) [reply SETUP.md]: Test the python and it works. Updating branch with new commands.

File: SETUP.md [hunk 3]
```diff
@@ -97,11 +97,11 @@ The following command automates this substitution.
 
 ```sh
 # Edit ci/deploy.sh with travis secure env variables generated by travis encrypt-file
-TRAVIS_ENCRYPT_ID=`grep \
-  --only-matching --perl-regexp \
-  --regexp='(?<=encrypted_)[a-zA-Z0-9]+(?=_key)' \
-  travis-encrypt-file.log`
-sed --in-place "s/f2f00aaf6402/$TRAVIS_ENCRYPT_ID/g" deploy.sh
+TRAVIS_ENCRYPT_ID=`python -c "import re; \
+    print(re.search('encrypted_([a-zA-Z0-9]+)_key', \
```
[the remark this answers is not in the document]
Comment (dhimmel) [reply SETUP.md]: I wonder if readability would improve with the following method:

```python
import re;
text = open('travis-encrypt-file.log').read();
print(re.search('encrypted_([a-f0-9]+)_key', text).group(1))
```

File: SETUP.md [hunk 4]
```diff
@@ -97,11 +97,10 @@ The following command automates this substitution.
...
 ```sh
 # Edit ci/deploy.sh with travis secure env variables generated by travis encrypt-file
-TRAVIS_ENCRYPT_ID=`grep \
-  --only-matching --perl-regexp \
-  --regexp='(?<=encrypted_)[a-zA-Z0-9]+(?=_key)' \
-  travis-encrypt-file.log`
-sed --in-place "s/f2f00aaf6402/$TRAVIS_ENCRYPT_ID/g" deploy.sh
+TRAVIS_ENCRYPT_ID=`python -c "import re; \
+                  text = open('travis-encrypt-file.log').read(); \
+                  print(re.search('encrypted_([a-f0-9]+)_key', text).group(1))"`
```
Comment (dhimmel) [inline SETUP.md]: Our style for the shell commands is 2 space indent

Comment (vsmalladi) [reply SETUP.md]: Okay will update accordingly

**Comment by dhimmel** [comment]
Thanks @vsmalladi. I tested the commands locally on Linux (not on a fully new manubot instances, but in a simulation). They worked on Ubuntu 17.10

**Comment by dhimmel** [comment]
For future testing, here is an example `ci/travis-encrypt-file.log`:

```
encrypting deploy.key for greenelab/manubot-rootstock
storing result as deploy.key.enc
storing secure env variables for decryption

openssl aes-256-cbc -K $encrypted_1b0a5c5e82ff_key -iv $encrypted_1b0a5c5e82ff_iv -in deploy.key.enc -out deploy.key -d

Make sure to add deploy.key.enc to the git repository.
Make sure not to add deploy.key to the git repository.
Commit all changes to your .travis.yml.
```

I believe this was created with the `travis` gem v1.8.8.

**Pull request merged**